tests/neo.UnitTests/SmartContract/Native/UT_PolicyContract.cs

```diff
@@ -231,21 +235,6 @@ public void TestCheckPolicy()
         {
             Transaction tx = Blockchain.GenesisBlock.Transactions[0];
             var snapshot = Blockchain.Singleton.GetSnapshot();
-
```

Comment:
@erikzhang @eryeer @shargon, double check this because perhaps this test was wrong.

Reply:
How about change it like this.
```C#
        [TestMethod]
        public void TestCheckPolicy()
        {
            Transaction tx = Blockchain.GenesisBlock.Transactions[0];
            var snapshot = Blockchain.Singleton.GetSnapshot();
            StorageKey storageKey = new StorageKey
            {
                ScriptHash = NativeContract.Policy.Hash,
                Key = new byte[sizeof(byte)]
            };
            storageKey.Key[0] = 15;
        
            var item = snapshot.Storages.GetAndChange(storageKey);
            item.Value = new UInt160[] { tx.Sender }.ToByteArray();
            NativeContract.Policy.CheckPolicy(tx, snapshot).Should().BeFalse();

            snapshot = Blockchain.Singleton.GetSnapshot();
            NativeContract.Policy.CheckPolicy(tx, snapshot).Should().BeTrue();
        }
```

Reply:
I tried, it says `True` instead of `False`, that is why I believe the test was wrong. Why was it `False` in any situation?

Reply:
Using this code, the test will pass. We need to change the return value of `snapshot.Storages.GetAndChange(storageKey)` instead of passing a StorageItem creating function into it.

Reply:
Anhhh, I see, but it would be better to DeleteKey, no?

Reply:
As long as we don't commit the snapshot and don't change the original storage , I think any kind of operation is fine. 😄

Reply:
@eryeer, but why would this be `False`:
```cs
        internal bool CheckPolicy(Transaction tx, StoreView snapshot)
        {
            UInt160[] blockedAccounts = GetBlockedAccounts(snapshot);
            if (blockedAccounts.Intersect(tx.GetScriptHashesForVerifying(snapshot)).Any())
                return false;
            return true;
        }
```

I do not see the connection with the `BlockedAccounts`?

Reply:
`storageKey.Key[0] = 15;` this key is the blockedAccount key
https://github.com/neo-project/neo/blob/0f75741e2ec471235cba99b6ffc6b38b634dd47e/src/neo/SmartContract/Native/PolicyContract.cs#L24
Because it is private, we can only use the 15 instead of `Prefix_BlockedAccounts`

Reply:
I partially understood! Thanks for the explanation, @eryeer.

However, I still need to figure out this part:

> We need to change the return value of snapshot.Storages.GetAndChange(storageKey) instead of passing a StorageItem creating function into it. Because the initalization will add a storageItem into storage, trackable.Item won't be null.

Reply:
😄 Here is the initalization.
https://github.com/neo-project/neo/blob/0f75741e2ec471235cba99b6ffc6b38b634dd47e/src/neo/SmartContract/Native/PolicyContract.cs#L62-L65
So whenever we get the BlockedAccounts from Internal storage, the value will be new `UInt160[0].ToByteArray()` by default.

Reply:
I think that I understood.

This would also work:

```cs
            Transaction tx = Blockchain.GenesisBlock.Transactions[0];
            var snapshot = Blockchain.Singleton.GetSnapshot();
            StorageKey storageKey = new StorageKey
            {
                ScriptHash = NativeContract.Policy.Hash,
                Key = new byte[sizeof(byte)]
            };
            storageKey.Key[0] = 15;
        
            //var item = snapshot.Storages.GetAndChange(storageKey);
            //item.Value = new UInt160[] { tx.Sender }.ToByteArray();
            snapshot.Storages.Delete(storageKey);
            snapshot.Storages.GetAndChange(storageKey, () => new StorageItem
            {
                Value = new UInt160[] { tx.Sender }.ToByteArray()
            });
            snapshot.Commit();
            NativeContract.Policy.CheckPolicy(tx, snapshot).Should().BeFalse();
            snapshot = Blockchain.Singleton.GetSnapshot();

            snapshot.Storages.Delete(storageKey);
            snapshot.Storages.GetAndChange(storageKey, () => new StorageItem
            {
                Value = new UInt160[0].ToByteArray()
            });
            snapshot.Commit();

            NativeContract.Policy.CheckPolicy(tx, snapshot).Should().BeTrue();
```

Reply:
Oh yes, congratulations!